### PR TITLE
feat: add middleware setup

### PR DESCRIPTION
**Pull request by IgboPharaoh**
- This PR adds a middleware that checks if a route exists in the aliases object and maps it to the appropriate url.
Attends to issue #32

### CODE REVIEW

File: public/aliases.json
Comment (kouloumos) [inline public/aliases.json:1]: Nice implementation!
Do we need this file to be on source control?

File: src/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    return NextResponse.redirect(new URL(alias, req.url));
+  }
+  return NextResponse.next();
+}
+
+export const config = {
+  matcher: [
+    "/:path/:subpath*", // Exclude paths with file extensions
+    "/:path*/:file((?!.*\\.).*)", // Exclude specific Next.js paths and API routes
+    "/((?!api|_next/static|_next/image|favicon.ico|sitemap.xml|robots.txt|plugins).*)", // Matches any path with at least one subdirectory
```
Comment (Emmanuel-Develops) [inline src/middleware.ts:21]: The comments are in the wrong order

File: public/aliases.json
Comment (IgboPharaoh) [reply public/aliases.json:1]: We don't specifically since it's generated at build time, will remove

**Comment by kouloumos** [comment]
ACK, I verified that aliases are defined/redirected correctly in the PR deployment.

**Pull request merged**